socorro/cron/jobs/daily_url.py

```diff
@@ -2,103 +2,329 @@
+    public_out_directory = config.get('public_output_path')
+    public_gzip_file_handle = None
+    public_csv_file_handle = None
+    if public_out_directory:
+        public_out_pathname = os.path.join(
+            public_out_directory,
+            pubic_out_filename
+        )
+        public_gzip_file_handle = gzip.open(public_out_pathname, "w")
+        public_csv_file_handle = csv.writer(
+            public_gzip_file_handle,
+            delimiter='\t',
+            lineterminator='\n'
+        )
+    else:
+        config.logger.info("Will not create public (bowdlerized) gzip file")
+    config.public_out_pathname = public_out_pathname
+    yield (private_csv_file_handle, public_csv_file_handle)
+    private_gzip_file_handle.close()
+    if public_gzip_file_handle:
+        public_gzip_file_handle.close()
+
+
+#------------------------------------------------------------------------------
+linux_line_re = re.compile(
+    r'0\.0\.0 [lL]inux.+[lL]inux$|[0-9.]+.+(i586|i686|sun4u|i86pc|x86_64)?'
+)
+linux_version_re = re.compile(
+    r'(0\.0\.0 [lL]inux.)([0-9.]+[0-9]).*(i586|i686|sun4u|i86pc|x86_64).*'
+)
+linux_short_version_re = re.compile(r'(0\.0\.0 [lL]inux.)([0-9.]+[0-9]).*')
 
 
+#------------------------------------------------------------------------------
+def get_appropriat_os_version(name, original_version):
+    """
+    If this is a linux os, chop out all the gubbish retaining only the actual
+    version numbers and, if available, the architecture name.
+    """
+    ret = original_version
+    if 'Linux' != name:
+        pass
+    elif not linux_line_re.match(original_version):
+        ret = ''
+    else:
+        m = linux_version_re.sub(r'\2 \3', original_version)
+        ret = m
+        if original_version == m:
+            m = linux_short_version_re.sub(r'\2', original_version)
+            ret = "%s ?arch?" % (m)
+            if original_version == m:
+                ret = ''
+    return ret
+
+
+#------------------------------------------------------------------------------
+def process_crash(a_crash_row):
+    column_value_list = []
+    os_name = None
+    ooid = ''
+    for i, x in enumerate(a_crash_row):
+        if x is None:
+            x = r'\N'
+        if i == 2:
+            ooid = x.rsplit('/', 1)[-1]  # existing Bug - unused
+        if i == 10:  # r.os_name
+            x = os_name = x.strip()
+        if i == 11:  # r.os_version
+            # per bug 519703
+            x = get_appropriat_os_version(os_name, x)
+            os_name = None
+        if i == 14:  # bug_associations.bug_id
+            x = ','.join(str(bugid) for bugid in x)
+        if i == 15:  # r.user_comments
+            x = x.replace('\t', ' ')  # per bug 519703
+        if i == 17:  # r.email -- show 'email' if the email is likely useful
+            # per bugs 529431/519703
+            if '@' in x:
+                x = 'yes'
+            else:
+                x = ''
+        if type(x) == str:
```

Comment:
`if isinstance(x, basestring)` in case we want to be compatible with unicode strings.